src/update-helper/update-helper.c

```diff
@@ -0,0 +1,969 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "sd-bus.h"
+#include "sd-daemon.h"
+
+#include "ansi-color.h"
+#include "build.h"
+#include "bus-error.h"
+#include "bus-locator.h"
+#include "bus-unit-util.h"
+#include "bus-util.h"
+#include "bus-wait-for-jobs.h"
+#include "fileio.h"
+#include "install.h"
+#include "log.h"
+#include "macro.h"
+#include "main-func.h"
+#include "path-util.h"
+#include "string-util.h"
+#include "strv.h"
+#include "unit-name.h"
+#include "user-util.h"
+#include "verbs.h"
+
+static RuntimeScope arg_runtime_scope = RUNTIME_SCOPE_SYSTEM;
+static bool arg_quiet = false;
+static bool arg_stdin = false;
+static bool arg_dry_run = false;
+static bool arg_removed = false;
+
+static int help(void) {
+        printf("%1$s [OPTIONS...] COMMAND [UNITS...]\n\n"
+               "%2$sHelper tool for package manager integration.%3$s\n"
+               "\n%2$sOptions:%3$s\n"
+               "  -h --help                         Show this help\n"
+               "     --version                      Show version\n"
+               "     --system                       Operate on system manager\n"
+               "     --global                       Operate on user managers\n"
+               "     --quiet                        Suppress error logging in some scenarios\n"
+               "     --stdin                        Read unit file paths from standard input\n"
+               "     --dry-run                      Only print what would be done\n"
+               "     --removed                      Operate on units that were removed on disk\n"
+               "\n%2$sCommands:%3$s\n"
+               "  install-units UNIT...             Enable and preset units\n"
+               "  remove-units UNIT...              Disable and stop units\n"
+               "  mark-restart-units UNIT...        Mark units for restart\n"
+               "  mark-reload-units UNIT...         Mark units for reload\n"
+               "  reload                            Reload manager configuration\n"
```

Comment:
I am not a fan that "reload" here means something quite different from what it means in systemctl... 

hence, rename this daemon-reload?